Make taiko use taiko-specific hitsounds

smoogipoo opened the pull request:
Fixes #1765 also.

smoogipoo committed:
Give SampleControlPoint a 100% volume by default

smoogipoo committed:
Make ControlPointInfo return a set default control point

smoogipoo committed:
Add test case for InputDrum

smoogipoo committed:
Make taiko use namespaced hitsounds

This is a temporary solution for now that uses DrawableHitObject.SampleNamespace for the override. We will not want to do this going forward, and instead have the rulesets add their custom resource stores to the games', but that requires deciding where/when to apply/remove such resource stores, and is probably left to skinning.

smoogipoo committed:
Taiko drumroll ticks should just play the playfield samples

smoogipoo committed:
Fix spelling error

peppy commented on osu.Game/Beatmaps/ControlPoints/ControlPointInfo.cs:
```diff
@@ -12,6 +12,11 @@ namespace osu.Game.Beatmaps.ControlPoints
     [Serializable]
     public class ControlPointInfo
     {
+        private static readonly TimingControlPoint default_timing_point = new TimingControlPoint();
```
let's hope no one changes these by accident >.>

smoogipoo replied:
Yeah I considered that too, these should definitely be readonly or structs.

peppy replied:
is this just a preoptimisation? or is there another reason for it?

smoogipoo replied:
Could attempt but not sure how big of a change

peppy replied:
I mean what's the reason for static in the first place?

smoogipoo replied:
Actually you bring up a good point, I could override GetHashCode instead... It's because these are used for dictionary keys.

smoogipoo committed:
Revert "Make ControlPointInfo return a set default control point"

This reverts commit 3113a55a988be6fe7112f73d1c905edd0cd787ab.

smoogipoo committed:
Lookup by control point time rather than control point

Under some situations, such as when there are no control points, ControlPointInfo will return to us a newly-constructed controlpoint with every call to SamplePointAt, which will fail on this key lookup.

Between this and overriding GetHashCode, I think this is the more proper fix for this.

smoogipoo committed:
Merge remote-tracking branch 'origin/master' into proper-taiko-hitsounds

smoogipoo committed:
Merge branch 'master' into proper-taiko-hitsounds

smoogipoo committed:
Update framework

peppy committed:
Merge branch 'master' into proper-taiko-hitsounds

peppy committed:
Simplify path construction

peppy committed:
Add better sample fallback logic

Also adds support for null channels at InputDrum level.

peppy committed:
Merge branch 'master' into proper-taiko-hitsounds

peppy committed:
Merge branch 'master' into proper-taiko-hitsounds
